cocotb/drivers/amba.py

```diff
@@ -180,6 +184,8 @@ def read(self, address, sync=True):
 
         raise ReturnValue(data)
 
+    def __len__(self):
+        return 2**len(self.bus.ARADDR)
```

Comment:
`AXI4LiteMaster` does not implement `__getitem__` or `__setitem__`, so this seems like an abuse of python's `__len__` to me.

If you want to expose this information somehow, perhaps better would be:
```python
@property
def addresses(self):
    # make sure we use the lazy form here
    if sys.version_info.major == 2:
        range = xrange
    return range(0, 2**len(self.bus.ARADDR))
```